### DISCUSSION TITLE
Whats the best way to generate focus colors in v4 ?

### DISCUSSION THREAD

**Discussion by technophile-04** [saadeghi/daisyui#2647] [Q&A] [answered]
### Description : 

It seems that in v4 we are no more generating `--sf` and `--pf` vars and instead we are just making the colors of buttons more black instead of making a color bit darker tone which we used to do in dasiy UI v3 : 

https://github.com/saadeghi/daisyui/assets/80153681/07da9b55-69ca-4501-8664-c9c4ef79bb20

Checkout how left v4 just becomes more black instead of changing the tone a bit on v3 (right)

Docs suggested to something like this inorder to get `focus` colors : 

`background-color: color-mix(in oklab,oklch(var(--p)), black 7%)`

but this doesn't create a color similar to `--pf` instead it just add more black similar to what we have currently in v4 already :(

**Comment by saadeghi** [comment, marked as answer]
I understand that you were preferring the old hover color but that saturation was not intended!

It's a problem with HSL color system. When we change the Lightness of a color, saturation doesn't stay the same and it changes in a random number (depending on the Hue!) and it creates a big inconsistency for different colors. What we wanted was to change the lightness to a fixed amount but it was making some colors of some themes less saturated and some colors of some themes more saturated. Because the way HSL color model works…

For example, in daisyUI 3, this was the color of `btn-primary` in `synthwave` theme:
```
hsl(321 70% 69%)
```
And this was the hover color:
```
hsl(321 70% 62%)
```
<img width="295" alt="Screenshot 2023-12-12 at 8 47 08 PM" src="https://github.com/saadeghi/daisyui/assets/7342023/a5f72c3b-cd48-48cf-892d-c5c340196d21">
<img width="295" alt="Screenshot 2023-12-12 at 8 47 27 PM" src="https://github.com/saadeghi/daisyui/assets/7342023/d13e387f-4f30-4271-b5e5-37cbaa0d5ea5">

As you can see we were only changing Lightness value by 7% to make it darker but that was causing the Saturation to change visually even though the Saturation value is the same!

This inconsistency was uncontrollable for different colors using HSL color system. Some colors were getting more saturated, some colors less saturated.  

In daisyUI 4, we have consistent color change for button hover. Using CSS `color-mix()` we mix a color with black which results a similar effect on all colors. It's like putting transparent dark overlay on the color.

This `color-mix()` style currently is used for `btn` hover style and `link` hover style. And of course if you don't like this behavior, you can override it using `hover:bg-[some_color]` utility class, either in HTML or in CSS file.

Let me know if you have a question.

**Reply by technophile-04** [reply]
Ohh makes sense! Tysm @saadeghi for the thorough response,  really appreciate it 🙌
